Cargo.toml

```diff
@@ -24,9 +28,9 @@ serde = ["dep:serde"]
 scale-info = ["dep:scale-info"]
 
 [dependencies]
-scale-info = { version = "2.2.0", default-features = false, features = ["std"], optional = true }
-codec = { package = "parity-scale-codec", version = "3.2.0", default-features = false, features = [] }
-serde = { version = "1", optional = true }
+scale-info = { version = "2.2.0", default-features = false, optional = true }
```

Comment:
Let's bump this to 2.5.0 or above, since you're using the pub fields now which requires at least that version :)